### PR TITLE
Add topic framework for new content

### PR DESCRIPTION
**Pull request by SueChaplain**
Adding a simple framework of topics
to supplement the command reference.
Currently these contain minimal
content, which will be built up
over time from contributions.

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@pshipton @DanHeidinga @mstoodle 
Work to add new content into the OpenJ9 user guide has to start somewhere. So I've created some "scaffolding" that can be contributed to over the weeks to come. I'm hoping that you are good with this approach - the sections are pretty minimal, but when I look at 0.10.0 items like https://github.com/eclipse/openj9/issues/1932, the content will need somewhere to go.

As everything new is re-written I'd like someone to go through and make sure there are no glaring technical errors or omissions.

Built HTML can be viewed on http://staging.eclipse.org/openj9/docs/ if that is easier than viewing in GitHub (not all the markup will render).

File: docs/aot.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Ahead-Of-Time (AOT) compiler
+
+The AOT compiler dynamically compiles Java methods into native AOT code at runtime and stores them in the shared classes cache. This activity enables the VM to start an application faster the next time it runs because it doesn't need to spend time interpreting Java methods. The VM automatically chooses which methods should be AOT-compiled based on heuristics that identify the start-up phase of large applications. AOT code is always used in combination with class data sharing and is enabled automatically when `-Xshareclasses` is set on the command line. When a cached AOT method is run it might also be optimized further by the Just-In-Time (JIT) compiler.
+
+If you want to turn off AOT compilation, set the `-Xshareclasses:noaot` suboption. When the AOT compiler is disabled, the JIT compiles frequently used methods into native code. However, because the JIT compiler operates while the application is running, the startup time for an application will increase.
```
Comment (pshipton) [inline docs/aot.md]: Note `-Xshareclasses:noaot` is defined as "Disables caching and loading of AOT code. AOT code already in the shared data cache can be loaded."

There is also a `-Xnoaot` option, which is defined as "This option turns off the AOT compiler and disables the use of AOT-compiled code."

File: docs/dump_heapdump.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Heap dumps contain a snapshot of all the live objects that are being used by a running Java application on the Java heap. There are two formats for heap dumps; the classic format, which is ascii text and the PHD format, which is compressed and not readable.
+
+A heap dump contains a list of all object instances. For each object instance you can find the following additional data:
+
+- The object address
+- The type or class name
+- The size
+- Any references to other objects
+
+For a visual analysis of a heap dump as an aid to problem determination, use the [Eclipse Memory Analyzer tool (MAT)](https://www.eclipse.org/mat/) or the [IBM Memory Analyzer tool](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_8.0.0/com.ibm.java.80.doc/diag/tools/tool_memoryanalyzer.html), which extends MAT by using the Diagnostic Tool Framework for Java (DTFJ)
```
Comment (pshipton) [inline docs/dump_heapdump.md]: For MAT to work properly with OpenJ9, you need to install the the DTFJ plugin.

`Install the DTFJ plugin using Help > Install New Software > Work with "IBM Diagnostic Tool Framework for Java" > IBM monitoring and  Diagnostic Tools > Diagnostic Tool Framework for Java`

With the DTFJ plugin, MAT can also examine system dumps.

File: docs/dump_systemdump.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# System dump
+
+System dumps, often known as *core dumps*, are platform-specific and contain a raw binary dump of the process memory. This type of dump has a complete copy of the Java heap, including the contents of all Java objects in the application.
+
+To examine a system dump you can use the OpenJ9 dump viewer (`jdmpview`) or a platform-specific debugging tool.
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** On Linux&trade; and AIX&reg; systems you must use the OpenJ9 dump extractor (`jextract`) to collect additional files for analysis before you can view the dump.
```
Comment (pshipton) [inline docs/dump_systemdump.md]: This is no longer accurate. There is no requirement to use jextract before using jdmpview (note in the IBM doc the statement about using jextract is qualified with "if the system dump was generated by a version of the J9 virtual machine before V2.6").

The only purpose for jextract for OpenJ9 is to collect system libraries on LKinux or AIX when engaging the IBM service organization. https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.vm.80.doc/docs/systemdump_viewer_jextract.html

File: docs/jit.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+The JIT compiler is enabled by default to optimize performance. However, if you experience a problem running your application, temporarily turning off the JIT will tell you whether the JIT is at fault.
+
+Because JIT starts at the same time as the VM, you can only modify JIT behavior at startup.
+
+There are a number of ways to disable the JIT:
+
+- Specify `-Djava.compiler=NONE` on the command line.
+- Specify `-Xint` on the command line, which turns off the JIT and AOT compiler. To eliminate problems with one or the other you can turn these compilers off selectively with the `-Xnojit` and `-Xnoaot` options.
+- Call the `java.lang.Compiler` API programmatically.
```
Comment (pshipton) [inline docs/jit.md]: Note the java.lang.Compile class is deprecated for removal since Java 9.

File: docs/aot.md (hunk 1 above)
Comment (SueChaplain) [reply docs/aot.md]: Switched the options over. Thanks

File: docs/dump_heapdump.md (hunk 2 above)
Comment (SueChaplain) [reply docs/dump_heapdump.md]: Incorporated your updates. Thanks

File: docs/dump_systemdump.md (hunk 3 above)
Comment (SueChaplain) [reply docs/dump_systemdump.md]: I've removed the note about jextract, and the topic in the Tools section. Now I'm looking at the Dump viewer topic and seeing the need for some changes there too (in the IBM doc aswell). I can remove references here to jextract, but should I also remove all the references to passing in "zip" on the command line when running jdmpview. There a quite a few of these, so checking first.

File: docs/jit.md (hunk 4 above)
Comment (SueChaplain) [reply docs/jit.md]: Fixed here.

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/dump_systemdump.md (hunk 3 above)
Comment (SueChaplain) [reply docs/dump_systemdump.md]: Done. I also removed -notemp as that also seemed redundant judging by the description.

File: docs/jit.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+The JIT compiler is enabled by default to optimize performance. However, if you experience a problem running your application, temporarily turning off the JIT will tell you whether the JIT is at fault.
+
+Because JIT starts at the same time as the VM, you can only modify JIT behavior at startup.
+
+There are a number of ways to disable the JIT:
+
+- Specify `-Djava.compiler=NONE` on the command line.
+- Specify `-Xint` on the command line, which turns off the JIT and AOT compiler. To eliminate problems with one or the other you can turn these compilers off selectively with the `-Xnojit` and `-Xnoaot` options.
+- Call the `java.lang.Compile` API programmatically.
```
Comment (pshipton) [inline docs/jit.md]: It should be java.lang.Compiler not Compile

Comment (SueChaplain) [reply docs/jit.md]: Fixed

**Pull request merged**